crates/ra_hir/src/code_model.rs

```diff
@@ -193,14 +193,32 @@ impl ModuleDef {
             ModuleDef::Function(it) => Some(it.name(db)),
             ModuleDef::EnumVariant(it) => Some(it.name(db)),
             ModuleDef::TypeAlias(it) => Some(it.name(db)),
-
             ModuleDef::Module(it) => it.name(db),
             ModuleDef::Const(it) => it.name(db),
             ModuleDef::Static(it) => it.name(db),
 
             ModuleDef::BuiltinType(it) => Some(it.as_name()),
         }
     }
+
+    pub fn resolver<D: DefDatabase + HirDatabase>(&self, db: &D) -> Option<Resolver> {
+        Some(match self {
+            ModuleDef::Module(m) => Into::<ModuleId>::into(m.clone()).resolver(db),
```

Comment:
Is there a reason this doesn't work? I find that much easier to read.
```suggestion
            ModuleDef::Module(m) => ModuleId::from(m.clone()).resolver(db),
```